### PR TITLE
Feat/challenge 03 loops conditionals

### PR DESCRIPTION
**Pull request by HuevoPlatino**

### CODE REVIEW

File: .prettierignore
Comment (tictools) [inline .prettierignore:1]: This file is empty.

⚠️ FIX: add `node_modules`

File: challenge_03_loops-conditionals/loops-conditionals/CHALLENGE.md
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/CHALLENGE.md:1]: `README.md` should only be defined at root level (in fact, it is already there)

⚠️ FIX: remove `README.md` from this folder and leave only the root one.

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_00.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    6. If it is neither "cat" nor "dog", print out "Must be an alien"
+    7. At the end of the program print out: "--- Animal check complete ---"
+    8. Change animal to different values to make sure it works in all cases
+*/
+
+console.log("--- Animal Checker Program ---");
+
+let animal = "dog";
+
+if (animal === "cat") {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_00.js]: ⚠️ FIX: Change animal to different values to make sure it works in all cases

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    Print result.
+
+    HINT: modulus operator ( % ) is your friend
+*/
+
+console.log("--- Even or Odd ---");
+
+let number = 4;
+
+if (number % 2 === 0) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js]: ⚠️ FIX: Change the variable to test it works for different cases. Print result.

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
+    1. Create a for loop that will print out all the even numbers between 10 and 40.
+    
+    2. Create a for loop that will print out all the odd numbers between 10 and 40.
+*/
+
+console.log("---Exercise 0---");
+// 1.
+for (let i = 10; i <= 40; i += 2) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js]: These loops should iterate over all numbers in between 10 and 40, printing only the odd/even ones, depending on its own logic.

⚠️ FIX: iterate over all numbers and print only the odd/even ones

You should take in account what variables are storing in order to name them properly. Is variable `i` referring to any index in this case?

⚠️ FIX: rename properly `for` variables in both cases

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    1. Create a for loop that loops while index is less or equal than 10.
+    2. Each loop:
+        * create a new random number between 1 and 10 (hint: check documentation regarding Math.floor and Math.random methods)
+        * evaluate if random number is greater than 5
+        * if it is equal or greater than 5 print "Logged In!"
+        * if it is less than 5 print "Logged Out!"
+*/
+
+console.log("---Exercise 1---");
+for (let i = 1; i <= 10; i++) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js]: ⚠️ FIX: rename properly `for` variables in both cases

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    Checkpoint! 40
+    Half way there!
+    Checkpoint! 60
+    Checkpoint! 70
+    Checkpoint! 80
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
+console.log("---Exercise 2---");
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js:21]: This is the output of your script...
```
---Exercise 2---
Checkpoint! 10
Checkpoint! 20
Checkpoint! 30
Checkpoint! 40
Checkpoint! 50
Half way there!
Checkpoint! 60
Checkpoint! 70
Checkpoint! 80
Checkpoint! 90
Checkpoint! 100
You made it!
```

...have you tested it before pushing? 🤔

⚠️ FIX: needs work.

⚠️ FIX: rename properly `for` variable

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_00.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    4. After the loops is done print out: "Sucessfully logged in!"
+
+    HINT1: Be careful for an infinite loop!
+*/
+
+console.log("---Exercise 00---");
+
+let loggedIn = false;
+
+while (loggedIn === false) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_00.js]: This is the output of your script...
```
Incorrect login credentials
Successfully logged in!
```

...have you tested it before pushing? 🤔

⚠️ FIX: needs work.

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    2. Create a while loop that will print out all the odd numbers between 10 and 40.
+*/
+
+console.log("---Exercise 01---");
+
+let num = 10;
+
+console.log("Even numbers between 10 and 40:");
+
+while (num <= 40) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_01.js:13]: These loops should iterate over all numbers in between 10 and 40, printing only the odd/even ones, depending on its own logic.

⚠️ FIX: iterate over all numbers and print only the odd/even ones

File: challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    Checkpoint! 90
+    You made it!
+    All, done!
+*/
+
+console.log("---Exercise 3---");
+
+let i = 1;
+
+while (i <= 100) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/while/exercise_02.js:26]: This is the output of your script...
```
---Exercise 3---
Checkpoint! 10
Checkpoint! 20
Checkpoint! 30
Checkpoint! 40
Checkpoint! 50
Half way there!
Checkpoint! 60
Checkpoint! 70
Checkpoint! 80
Checkpoint! 90
Checkpoint! 100
You made it!
All, done!
```

...have you tested it before pushing? 🤔

⚠️ FIX: needs work.

You should take in account what variables are storing in order to name them properly. Is variable `i` referring to any index in this case?

⚠️ FIX: rename properly `i` variable

**Review by tictools — CHANGES_REQUESTED** [review]
`node_modules` folder should never be pushed to remote repository.

In order to remove it, follow these steps:
* remove `node_modules` folder in your remote repository (branch `feat/challenge_03_loops-conditionals`)
* move to your local repo (same branch) and execute `git pull`
* fix requested changes and push them.

File: challenge_02_variable_operators/index (1).html
Comment (tictools) [inline challenge_02_variable_operators/index (1).html:1]: FIX: remvoe `(1)` from  file name

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_00.js
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_00.js:1]: Your are duplicating code and this is not a good idea... You should encapsulate this logic
```js
 if (animal === "cat") {
   console.log("Meow!");
 } else if (animal === "dog") {
   console.log("Woof!");
 } else {
   console.log("Must be an alien");
 }
```
inside a function which returns the expected string.

Then call it and log its returned value.

FIX: encapsulate logic

File: challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/conditionals/exercise_01.js:1]: Your are duplicating code and this is not a good idea... You should encapsulate this logic
```js
if (number % 2 === 0) {
  console.log("Even");
} else {
  console.log("Odd");
}
```
inside a function which returns the expected string.

Then call it and log its returned value.

FIX: encapsulate logic

**Review by tictools — CHANGES_REQUESTED** [review]

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_00.js:1]: You must iterate over all numbers between 10 and 40 and log even or odd ones, depending on the case. 
Do not increase iterations by 2.

FIX: iterate over all numbers.

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    2. Each loop:
+        * create a new random number between 1 and 10 (hint: check documentation regarding Math.floor and Math.random methods)
+        * evaluate if random number is greater than 5
+        * if it is equal or greater than 5 print "Logged In!"
+        * if it is less than 5 print "Logged Out!"
+*/
+
+console.log("---Exercise 1---");
+for (let index = 1; index <= 10; index++) {
+  let random = Math.floor(Math.random() * 10) + 1;
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_01.js:12]: FIX: stablish a new line between blocks
```suggestion
  let random = Math.floor(Math.random() * 10) + 1;
  
  if(...){
  //..
  }
```

File: challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+for (let index = 1; index <= 100; index++) {
+  if (index % 10 === 0 && index !== 50 && index !== 100) {
+    console.log("Checkpoint! " + index);
+  }
+  if (index === 50) {
```
Comment (tictools) [inline challenge_03_loops-conditionals/loops-conditionals/loops/exercise_02.js:26]: FIX: add new line between blocks

**Review by tictools — CHANGES_REQUESTED** [review]

File: .github/workflow/pipeline.yaml
Comment (tictools) [inline .github/workflow/pipeline.yaml:1]: You've got `.github/workflow/pipeline.yaml`

You should have `.github/workflows/pipeline.yaml` // worflows, not workflow

This check will never be executed

FIX: typo.

File: .gitingnore
Comment (tictools) [inline .gitingnore:1]: `.gitignore`, not`.gitingnore`

FIX: typo

**Review by tictools — CHANGES_REQUESTED** [review]

**Review by tictools — APPROVED** [review]

**Review by GALLEGOOOO — APPROVED** [review]

**Pull request merged**